clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -819,12 +833,12 @@ void call(Call call, long now) {
     public CreateTopicResults createTopics(final Collection<NewTopic> newTopics,
                                            final CreateTopicsOptions options) {
         final Map<String, KafkaFutureImpl<Void>> topicFutures = new HashMap<>(newTopics.size());
-        for (NewTopic newTopic : newTopics) {
-            topicFutures.put(newTopic.name(), new KafkaFutureImpl<Void>());
-        }
         final Map<String, CreateTopicsRequest.TopicDetails> topicsMap = new HashMap<>(newTopics.size());
         for (NewTopic newTopic : newTopics) {
-            topicsMap.put(newTopic.name(), newTopic.convertToTopicDetails());
+            if (topicFutures.get(newTopic.name()) == null) {
```

Comment:
Can't we just `put`? If there are duplicates (which is rare), we will simply overwrite, no? Same question for other places where we do the same.